src/components/__internal__/abstract-custom-field.jsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../icon/icon.jsx';
+import styles from './abstract-custom-field.css';
+import cautionSvg from '../../svgs/icon-caution-fill.svg';
+import FormControl from '../form-control/form-control.jsx';
+
+function getRootClassName(className, error, disabled) {
+  if (disabled) {
+    return `${className} ${styles.disabled}`;
+  }
+
+  if (error) {
+    return `${className} ${styles.error}`;
+  }
+
+  return className;
+}
+
+export default function AbstractCustomField(props) {
+  const labelId = `${props.id}-label`;
+  const invalidDueToProps = () => props.errorText.length > 0 && !props.readOnly;
+
+  const icon = () => {
+    if (invalidDueToProps()) {
+      return (<Icon
```

Comment:
This is going to conflict with @bennapp 's work to make multiple `Icon`'s show. Not sure if this should change here or there. Might come down to who goes green QA first. 😅

Reply:
Yeah, I think @bennapp and I will have to collaborate to get these pieces together